glean-core/src/metrics/uuid.rs

```diff
@@ -53,32 +55,31 @@ impl UuidMetric {
     /// ## Arguments
     ///
     /// * `glean` - The Glean instance this metric belongs to.
-    pub fn generate_and_set(&self, storage: &Glean) -> uuid::Uuid {
-        let uuid = uuid::Uuid::new_v4();
+    pub fn generate_and_set(&self, storage: &Glean) -> Uuid {
+        let uuid = Uuid::new_v4();
         self.set(storage, uuid);
         uuid
     }
 
-    /// Generate a new random UUID if none is stored yet.
+    /// Get the stored Uuid value.
```

Comment:
Let's call out that this is not expected to be user-facing API, but rather an internal thing for a specific purpose (state the purpose too :) ).

Reply:
IMO that's communicated by `pub(crate)`.